components/src/banner/banner.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, Input, OnInit, Output, EventEmitter } from "@angular/core";
+
+import { I18nService } from "jslib-common/abstractions/i18n.service";
+
+type BannerTypes = "premium" | "info" | "warning" | "danger";
+
+const defaultIcon: Record<BannerTypes, string> = {
+  premium: "bwi-star",
+  info: "bwi-info-circle",
+  warning: "bwi-exclamation-triangle",
+  danger: "bwi-error",
+};
+
+const defaultI18n: Partial<Record<BannerTypes, string>> = {
+  warning: "warning",
+  danger: "error",
+};
+
+@Component({
+  selector: "bit-banner",
+  templateUrl: "./banner.component.html",
+})
+export class BannerComponent implements OnInit {
+  @Input() type: BannerTypes = "info";
+  @Input() icon: string;
+  @Input() title: string;
+  @Input() useAlertRole = true;
```

Comment:
This is not referred to anywhere. Relatedly - you're missing the accessibility attributes referred to in the Figma.